### PR TITLE
Phase-1 Remove vavr dependency for CircuitBreaker and retry module

### PR DESCRIPTION
**Pull request by KrnSaurabh**
This Pull request is part of removing Vavr dependency from Resiliency4j- Phase 1.

This PR includes changes to make CircuitBreaker and Retry module entirely vavr independent. Changes are done to many other modules to back port the contract breaking changes done in core APIs.

My sincere apologies that the PR has become too long this time as I just couldn't keep track while back porting changes from core APIs.
I will try to keep the PRs smaller next time and iteratively remove Vavr from subsequent other modules.

@RobWin Let me know if the PR is too difficult to review. I will try to figure something out.

### CODE REVIEW

**Comment by RobWin** [comment]
Thanks. I have a look at it.

File: resilience4j-spring/src/main/java/io/github/resilience4j/ratelimiter/configure/RateLimiterConfiguration.java [hunk 1]
```diff
@@ -104,7 +104,7 @@ private RateLimiterRegistry createRateLimiterRegistry(
                         entry.getKey())));
 
         return RateLimiterRegistry.of(configs, rateLimiterRegistryEventConsumer,
-            io.vavr.collection.HashMap.ofAll(rateLimiterConfigurationProperties.getTags()));
+            rateLimiterConfigurationProperties.getTags());
```
Comment (RobWin) [inline resilience4j-spring/src/main/java/io/github/resilience4j/ratelimiter/configure/RateLimiterConfiguration.java:107]: I think we should create a defensive copy of the Map

File: resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerConfiguration.java [hunk 2]
```diff
@@ -142,8 +142,7 @@ CircuitBreakerRegistry createCircuitBreakerRegistry(
                     .createCircuitBreakerConfig(entry.getKey(), entry.getValue(),
                         customizerMap)));
 
-        return CircuitBreakerRegistry.of(configs, circuitBreakerRegistryEventConsumer,
-            io.vavr.collection.HashMap.ofAll(circuitBreakerProperties.getTags()));
+        return CircuitBreakerRegistry.of(configs, circuitBreakerRegistryEventConsumer, circuitBreakerProperties.getTags());
```
Comment (RobWin) [inline resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerConfiguration.java:145]: I think we should create a defensive copy of the Map

File: resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/threadpool/ThreadPoolBulkheadConfiguration.java [hunk 3]
```diff
@@ -102,8 +102,7 @@ private ThreadPoolBulkheadRegistry createBulkheadRegistry(
                 entry -> threadPoolBulkheadConfigurationProperties
                     .createThreadPoolBulkheadConfig(entry.getValue(),
                         compositeThreadPoolBulkheadCustomizer, entry.getKey())));
-        return ThreadPoolBulkheadRegistry.of(configs, threadPoolBulkheadRegistryEventConsumer,
-            io.vavr.collection.HashMap.ofAll(threadPoolBulkheadConfigurationProperties.getTags()));
+        return ThreadPoolBulkheadRegistry.of(configs, threadPoolBulkheadRegistryEventConsumer, threadPoolBulkheadConfigurationProperties.getTags());
```
Comment (RobWin) [inline resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/threadpool/ThreadPoolBulkheadConfiguration.java:105]: I think we should create a defensive copy of the Map

File: resilience4j-spring/src/main/java/io/github/resilience4j/timelimiter/configure/TimeLimiterConfiguration.java [hunk 4]
```diff
@@ -131,8 +131,7 @@ private static TimeLimiterRegistry createTimeLimiterRegistry(
                         entry -> timeLimiterConfigurationProperties.createTimeLimiterConfig(
                             entry.getKey(), entry.getValue(), compositeTimeLimiterCustomizer)));
 
-        return TimeLimiterRegistry.of(configs, timeLimiterRegistryEventConsumer,
-            HashMap.ofAll(timeLimiterConfigurationProperties.getTags()));
+        return TimeLimiterRegistry.of(configs, timeLimiterRegistryEventConsumer, timeLimiterConfigurationProperties.getTags());
```
Comment (RobWin) [inline resilience4j-spring/src/main/java/io/github/resilience4j/timelimiter/configure/TimeLimiterConfiguration.java:134]: I think we should create a defensive copy of the Map

File: resilience4j-spring/src/main/java/io/github/resilience4j/retry/configure/RetryConfiguration.java [hunk 5]
```diff
@@ -103,8 +103,7 @@ private RetryRegistry createRetryRegistry(
                     .createRetryConfig(entry.getValue(), compositeRetryCustomizer,
                         entry.getKey())));
 
-        return RetryRegistry.of(configs, retryRegistryEventConsumer,
-            io.vavr.collection.HashMap.ofAll(retryConfigurationProperties.getTags()));
+        return RetryRegistry.of(configs, retryRegistryEventConsumer, retryConfigurationProperties.getTags());
```
Comment (RobWin) [inline resilience4j-spring/src/main/java/io/github/resilience4j/retry/configure/RetryConfiguration.java:106]: I think we should create a defensive copy of the Map

**Review by RobWin — CHANGES_REQUESTED** [review]

**Comment by RobWin** [comment]
Thank you for your effort. Changes look very good!

**Review by RobWin — APPROVED** [review]

**Comment by KrnSaurabh** [comment]
@RobWin  can you also sync 2.0 branch with current master. I tried creating a PR to merge recent changes to 2.0 from master but I am getting some error and unable to create the Pull request.

**Pull request merged**